### PR TITLE
Brent Oakes - Code Challenge Pull Request

### PR DESCRIPTION
**Pull request by boakes**
## Overview

This is my Pull Request for the code-challenge. Below are some testing instructions to confirm that all requirements have been fulfilled. 

### Notes

If I had more time to work on this I would like to spend more time on error handling and make those responses have better pretty prints. 

## Testing Instructions

* How to test this PR
* Checkout this branch
* Manual test: 
* 1.  docker-compose build  
* 2.  docker-compose up
* 3. In a web browser go to: http://localhost:8000/
* 4. Test `123 main st chicago il` gets parsed to 
image
* 5. Test `123 main st chicago il 123 main st` returns an error. 
* 6. Confirm any additional addresses are parsed correctly/incorrectly 
* Automated test: docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app

### CODE REVIEW

File: parserator_web/static/js/index.js [hunk 1]
```diff
@@ -1,2 +1,50 @@
...
+
+        $.each(data['address_components'], function (key,value) {
+
+          tableData += '<tr>'
+          tableData += '<td>' + key + '</td>'
+          tableData += '<td>' + value + '</td>'
+          tableData += '</tr>'
+        })
+        addrResults.children('table').children('tbody').html(tableData)
+        //$('#address-table').children('tbody').html(tableData);
```
Comment (derekeder) [inline parserator_web/static/js/index.js]: we should remove commented out lines like these

File: parserator_web/static/js/index.js [hunk 2]
```diff
@@ -1,2 +1,50 @@
...
+        addrResults.show()
+      },
+      error: function (error)
+      {
+        //Would like to review this at some point to do some type 
+        //of pretty print for error text
```
Comment (derekeder) [inline parserator_web/static/js/index.js]: would you mind giving this a shot? I'd like to see what you had in mind. also, keep in mind the people using the parserator form are likely not programmers, so the output on the error message shouldn't be too technical

File: tests/test_views.py [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
+import json
+from django.urls import reverse
 
 
 def test_api_parse_succeeds(client):
-    # TODO: Finish this test. Send a request to the API and confirm that the
-    # data comes back in the appropriate format.
     address_string = '123 main st chicago il'
-    pytest.fail()
+    address_list = address_string.split(" ")
```
Comment (derekeder) [inline tests/test_views.py]: can you think of a way to test this using the `usaddress` library? remember, this test should be for the expected formatted output, not necessarily testing if usaddress is correct

**Review by derekeder — CHANGES_REQUESTED** [review]
@boakes thank you for submitting your code challenge. you did a great job organizing and testing your code.

I have a few suggested changes to make. Could you take a look and address ahead of our meeting next week?

File: parserator_web/static/js/index.js (hunk 1 above)
Comment (boakes) [reply parserator_web/static/js/index.js]: Oops! I think I removed them all now. Fixed in [b663459](https://github.com/datamade/code-challenge/pull/30/commits/b66345992783d979e6e132eeec8762c7de4a0a9d)

File: parserator_web/static/js/index.js (hunk 2 above)
Comment (boakes) [reply parserator_web/static/js/index.js]: I went ahead and did a simple alert box using some bootstrap css. I also made it so the full error message does still show up, but is hidden by default in an accordion card. Changed in [e299682](https://github.com/datamade/code-challenge/pull/30/commits/e2996827fbb0659c83d1b25804897c75f8a9c7cc)

File: tests/test_views.py (hunk 3 above)
Comment (boakes) [reply tests/test_views.py]: Is this more along the lines of what you were thinking? Changed in [81f508d](https://github.com/datamade/code-challenge/pull/30/commits/81f508d78116192920db5823481c878b131251ca)

**Pull request closed without merging**